bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,443 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wled.internal;
+
+import static org.openhab.binding.wled.internal.WLedBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SmartHomeUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WLedHandler} is responsible for handling commands and states, which are
+ * sent to one of the channels or http replies back.
+ *
+ * @author Matthew Skinner - Initial contribution
+ */
+
+@NonNullByDefault
+public class WLedHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final HttpClient httpClient;
+    private final WledDynamicStateDescriptionProvider stateDescriptionProvider;
+    private @Nullable ScheduledFuture<?> pollingFuture = null;
+    private BigDecimal masterBrightness255 = BigDecimal.ZERO;
+    private HSBType primaryColor = new HSBType();
+    private BigDecimal primaryWhite = BigDecimal.ZERO;
+    private HSBType secondaryColor = new HSBType();
+    private BigDecimal secondaryWhite = BigDecimal.ZERO;
+    private boolean hasWhite = false;
+    private WLedConfiguration config;
+
+    public WLedHandler(Thing thing, HttpClient httpClient,
+            WledDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        config = getConfigAs(WLedConfiguration.class);
```

Comment:
The `config` field shouldn't be initialized in the constructor. This should be done in `initialize()` instead. But it looks like you are already doing that so just remove it here.
```suggestion
```

Reply:
If I do that then I get "The @NonNull field config may not have been initialized", error by the compiler. If I make the config nullable then I have to check every time I use it, if it is null. So I went with this approach by doing it twice. How should it be handled then?

Reply:
The general pattern in other bindings is to initially set the config to a default instance. So in this case just define your config field as:
```
private WLedConfiguration config = new WLedConfiguration();
```

Reply:
FIXED: Thank you, its so obvious now you pointed it out.